src/bin/pgcopydb/extensions.c

```diff
@@ -200,31 +200,71 @@ copydb_copy_extensions_hook(void *ctx, SourceExtension *ext)
 		{
 			SourceExtensionConfig *config = &(ext->config.array[i]);
 
-			log_info("COPY extension \"%s\" "
-					 "configuration table \"%s\".\"%s\"",
-					 ext->extname,
-					 config->nspname,
-					 config->relname);
-
 			char qname[PG_NAMEDATALEN_FQ] = { 0 };
 
 			sformat(qname, sizeof(qname), "%s.%s",
 					config->nspname,
 					config->relname);
 
-			CopyArgs args = {
-				.srcQname = qname,
-				.srcAttrList = "*",
-				.srcWhereClause = config->condition,
-				.dstQname = qname,
-				.dstAttrList = "",
-				.bytesTransmitted = 0
-			};
-
-			if (!pg_copy(src, dst, &args))
+			/*
+			 * According to the PostgreSQL documentation, relkind 'r' is a
+			 * regular table, and 'S' is a sequence.
+			 * https://www.postgresql.org/docs/current/catalog-pg-class.html
+			 */
+			if (config->relkind == 'r')
+			{
+				CopyArgs args = {
+					.srcQname = qname,
+					.srcAttrList = "*",
+					.srcWhereClause = config->condition,
+					.dstQname = qname,
+					.dstAttrList = "",
+					.bytesTransmitted = 0
+				};
+
+				log_info("COPY extension \"%s\" "
+						 "configuration table %s",
+						 ext->extname,
+						 qname);
+
+				if (!pg_copy(src, dst, &args))
+				{
+					/* errors have already been logged */
+					return false;
+				}
+			}
+			else if (config->relkind == 'S')
+			{
+				SourceSequence seq = { };
+
+				strlcpy(seq.nspname, config->nspname, sizeof(seq.nspname));
+				strlcpy(seq.relname, config->relname, sizeof(seq.relname));
+				strlcpy(seq.qname, qname, sizeof(seq.qname));
+
+				log_info("COPY extension \"%s\" "
+						 "configuration sequence %s",
+						 ext->extname,
+						 qname);
+
+				if (!schema_get_sequence_value(src, &seq))
+				{
+					/* errors have already been logged */
+					return false;
+				}
+
+				if (!schema_set_sequence_value(dst, &seq))
+				{
+					/* errors have already been logged */
+					return false;
+				}
+			}
+			else
 			{
-				/* errors have already been logged */
-				return false;
+				log_warn("Extension \"%s\" configuration table %s "
+						 "is not a supported type '%c', skipping",
+						 ext->extname,
+						 qname,
+						 (char) config->relkind);
```

Comment:
Why is it a warning rather than an hard-error here? This needs some commenting if we want to confirm that choice, maybe with a link to Postgres docs with the relkind that are expected to be found in extension configuration setup?

Reply:
I made it as error and pointed to PG docs.